enumerables.rb

```diff
@@ -0,0 +1,139 @@
+module Enumerable
+  # 1. Create my_each
+  def my_each
+    (0..length - 1).each do |i|
+      yield self[i]
+    end
+  end
+
+  # 2. Create my_each_with_index
+  def my_each_with_index
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  # 3. Create my_select
+  def my_select
+    arr = []
+    (0..length - 1).each do |i|
+      criteria = yield(self[i])
+      arr << self[i] if criteria
+    end
+    arr
+  end
+
+  # 4. Create my_all?
+  def my_all?
+    arr = []
+    my_each do |i|
+      condition = yield(i)
+      arr << (condition ? true : false)
+    end
+    arr.include?(false) ? false : true
+  end
+
+  # 5. Create my_any?
+  def my_any?
+    arr = []
+    my_each do |i|
+      condition = yield(i)
+      arr << (condition ? true : false)
+    end
+    arr.include?(true)
+  end
+
+  # 6. Create my_none?
+  def my_none?
+    arr = []
+    my_each do |i|
+      condition = yield(i)
+      arr << (condition ? true : false)
+    end
+    arr.include?(true) ? false : true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
false_array.my_none? == true #true
```
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array.my_none?(String) == true
```
- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
words.my_none?(/z/) == words.none?(/z/)
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words.my_none?(5) == words.none?(5)
```